src/sc_pipe_management/input_params_generation/generate_terra_input_table.py

```diff
@@ -0,0 +1,597 @@
+import pandas as pd
+import os
+import requests
+import dataclasses
+import igvf_client
+import datetime
+import logging
+
+
+import sc_pipe_management.input_params_generation.constant as const
+import sc_pipe_management.input_params_generation.portal_metadata_parsing as portal_parsing
+import sc_pipe_management.input_params_generation.seqspec_parsing as seqspec_parsing
+
+
+def download_file_via_https(seqspec_file_url: str, igvf_api: igvf_client.api.igvf_api.IgvfApi, partial_root_dir: str) -> str:
+    """Download file via href url on the portal and save it locally."""
+    download_folder = os.path.join(partial_root_dir, 'temp')
+    if not os.path.exists(download_folder):
+        os.makedirs(download_folder)
+    igvf_api_config = igvf_api.api_client.configuration
+    session = requests.Session()
+    session.auth = (igvf_api_config.username, igvf_api_config.password)
+    response = session.get(seqspec_file_url)
+    if response.status_code == 200:
+        curr_output_file = os.path.join(
+            download_folder, seqspec_file_url.split('/')[-1])
+        with open(curr_output_file, 'wb') as file:
+            file.write(response.content)
+        return curr_output_file
+    else:
+        raise const.BadDataException(
+            f'Error: Download failed with status code: {response.status_code}.')
+
+
+def get_seqspec_accession_from_path(seqspec_file_path: str) -> str:
```

Comment:
function _get_seqspec_accession_from_path and get_seqspec_accession_from_path are duplicated